packages/@sanity/migrate/src/fetch-utils/__test__/assert2xx.test.ts

```diff
@@ -42,3 +42,22 @@ test('server responds with 5xx and no json response', () => {
     message: 'HTTP Error 500: Internal Server Error',
   })
 })
+
+test('server responds with 5xx and json response', () => {
+  const mockResponse = {
+    status: 500,
+    statusText: 'Internal Server Error',
+    json: () =>
+      Promise.resolve({
+        error: {
+          type: 'validationError',
+          description: 'Document is not of valid type',
+        },
+        status: 500,
+      }),
+  }
```

Comment:
If our env supports the response constructor, this may be nicer

```suggestion
  const mockResponse = new Response(
    JSON.stringify({
      error: {
        type: 'validationError',
        description: 'Document is not of valid type',
      },
      status: 500,
    }),
    {
      status: 500,
      statusText: 'Internal Server Error',
    },
  )
```

Reply:
probably worth doing it but maybe in a follow up PR, would need to change a few tests